src/proto/streams/flow_control.rs

```diff
@@ -25,32 +27,40 @@ fn sanity_unclaimed_ratio() {
 
 #[derive(Copy, Clone, Debug)]
 pub struct FlowControl {
-    /// Window size as indicated by the peer. This can go negative.
-    window_size: i32,
+    /// Window the peer knows about.
+    ///
+    /// This can go negative if a SETTINGS_INITIAL_WINDOW_SIZE is received.
+    ///
+    /// For example, say the peer sends a request and uses 32kb of the window.
+    /// We send a SETTINGS_INITIAL_WINDOW_SIZE of 16kb. The peer has to adjust
+    /// its understanding of the capacity of the window, and that would be:
+    ///
+    /// ```notrust
+    /// default (64kb) - used (32kb) - settings_diff (64kb - 16kb): -16kb
+    window_size: Window,
 
-    /// The amount of the window that is currently available to consume.
-    available: WindowSize,
+    /// Window that we know about.
+    ///
+    /// This can go negative if a user declares a smaller target window than
+    /// the peer knows about.
+    available: Window,
 }
 
 impl FlowControl {
     pub fn new() -> FlowControl {
         FlowControl {
-            window_size: 0,
-            available: 0,
+            window_size: Window(0),
+            available: Window(0),
         }
     }
 
     /// Returns the window size as known by the peer
     pub fn window_size(&self) -> WindowSize {
```

Comment:
Is there a reason that this fn doesn't return `Window` now instead of `WindowSize`?

Reply:
Because I wasn't sure of all the places `flow.window_size()` is used, and if it should return a possibly negative number, so I left that part alone...